GVFS/GVFS.Common/Database/PlaceholderTable.cs

```diff
@@ -0,0 +1,223 @@
+                    }
+
+                    return fileEntries;
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new GVFSDatabaseException($"{nameof(PlaceholderTable)}.{nameof(this.GetAllFilePaths)} Exception", ex);
+            }
+        }
+
+        public void AddPlaceholderData(IPlaceholderData data)
+        {
+            if (data.IsFolder)
+            {
+                if (data.IsExpandedFolder)
+                {
+                    this.AddExpandedFolder(data.Path);
+                }
+                else if (data.IsPossibleTombstoneFolder)
+                {
+                    this.AddPossibleTombstoneFolder(data.Path);
+                }
+                else
+                {
+                    this.AddPartialFolder(data.Path);
+                }
+            }
+            else
+            {
+                this.AddFile(data.Path, data.Sha);
+            }
+        }
+
+        public void AddFile(string path, string sha)
+        {
+            this.Insert(new PlaceholderData() { Path = path, PathType = PlaceholderData.PlaceholderType.File, Sha = sha });
+        }
+
+        public void AddPartialFolder(string path)
+        {
+            this.Insert(new PlaceholderData() { Path = path, PathType = PlaceholderData.PlaceholderType.PartialFolder });
+        }
+
+        public void AddExpandedFolder(string path)
+        {
+            this.Insert(new PlaceholderData() { Path = path, PathType = PlaceholderData.PlaceholderType.ExpandedFolder });
+        }
+
+        public void AddPossibleTombstoneFolder(string path)
+        {
+            this.Insert(new PlaceholderData() { Path = path, PathType = PlaceholderData.PlaceholderType.PossibleTombstoneFolder });
+        }
+
+        public void Remove(string path)
+        {
+            try
+            {
+                using (IDbConnection connection = this.connectionPool.GetConnection())
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = "DELETE FROM Placeholder WHERE path = @path;";
+                    command.AddParameter("@path", DbType.String, path);
+                    command.ExecuteNonQuery();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new GVFSDatabaseException($"{nameof(PlaceholderTable)}.{nameof(this.Remove)}({path}) Exception", ex);
+            }
+        }
+
+        private void Insert(PlaceholderData placeholder)
+        {
+            try
+            {
+                using (IDbConnection connection = this.connectionPool.GetConnection())
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = "INSERT OR REPLACE INTO Placeholder (path, pathType, sha) VALUES (@path, @pathType, @sha);";
+                    command.AddParameter("@path", DbType.String, placeholder.Path);

```

Comment:
From what I've see in SO posts I *think* SQLite is case-sensitive.  Do we need to worry about duplicate entries being added, or is there something about how VFS4G uses the database that prevents that?

Reply:
Another concern, calls to `Remove` using the current case of the file might fail to find existing entries that only differ by case.

Reply:
As discussed in a different comment, path in the table is now case insensitive, marking this as resolved.

Reply:
@wilbaker @kewillford does it need to be case-sensitive on OSes where the underlying filesystem is case-sensitive?

Reply:
It will but we have so many collections that are `OrdinalIgnoreCase` that there would need to be a bunch of code changes to make VFSForGit case sensitive on those file systems.